### PR TITLE
Estimate gas instead of hardcoded

### PR DESCRIPTION
**Pull request by marshall2112**
# Description
What does this PR solve?

Get an _actual_ gas estimate and use that for borrow, repay, repayall, supply, and withdraw.

# Checklist
- [ ] Code follows the style guide
- [ ] I have performed a self-review of my own code
- [ ] New and existing tests pass locally
- [ ] This PR is targeting the correct branch

### CODE REVIEW

**Comment by nichosystem** [comment]
Deploy preview for _core_ ready!

✅ Preview
https://core-21mjcyjq1-templedao.vercel.app
https://pr-1010-preview.templedao.link

Built with commit c64f2d73d34461251d1f4e1b79c994fd0d9fff61.
This pull request is being automatically deployed with [vercel-action](https://github.com/marketplace/actions/vercel-action)

File: apps/dapp/src/components/Pages/Core/DappPages/Borrow/index.tsx [hunk 1]
```diff
@@ -359,8 +362,12 @@ export const BorrowPage = () => {
       getTokenInfo(state.outputToken).decimals
     );
     try {
-      const tx = await tlcContract.borrow(amount, wallet, { gasLimit: 500000 });
-      const receipt = await tx.wait();
+      const preparedTransaction = await tlcContract.populateTransaction.borrow(
```
Comment (frost-ostrich) [inline apps/dapp/src/components/Pages/Core/DappPages/Borrow/index.tsx:365]: change `preparedTransaction` to `populatedTransaction` for naming consistency

File: apps/dapp/src/components/Pages/Core/DappPages/Borrow/index.tsx [hunk 2]
```diff
@@ -395,9 +402,14 @@ export const BorrowPage = () => {
         env.contracts.tlc,
         amount
       );
-      // Repay DAI
-      const tx = await tlcContract.repay(amount, wallet, { gasLimit: 400000 });
-      const receipt = await tx.wait();
+
+      // Note Repay vs RepayAll
+      const preparedTransaction = await tlcContract.populateTransaction.repay(
```
Comment (frost-ostrich) [inline apps/dapp/src/components/Pages/Core/DappPages/Borrow/index.tsx:407]: ditto `preparedTransaction` to `populatedTransaction`

**Review by frost-ostrich — APPROVED** [review]
LGTM % naming consistency

**Review by frost-ostrich — APPROVED** [review]

**Pull request merged**